lib/common/client/base.rb

```diff
@@ -119,10 +117,26 @@ def request(method, path, params = {}, headers = {}, options = {}) # rubocop:dis
                       else
                         Common::Client::Errors::ClientError
                       end
+        status, body = response_hash_from_error(e)[:response].try(:values_at, :status, :body)
+        raise error_class.new(e.message, status, body)
+      end
+
+      def response_hash_from_error(error)
+        return { response: nil } unless error.respond_to?(:response)
+
+        resp = error.response
+
+        if %i[status body headers].all? { |method| resp.respond_to? method }
+          return {
+            response: {
+              status: resp.status,
+              body: resp.body,
+              headers: resp.headers
+            }
+          }
+        end
 
-        response_hash = e.response&.to_hash
-        client_error = error_class.new(e.message, response_hash&.dig(:status), response_hash&.dig(:body))
-        raise client_error
+        { response: resp.try(:to_hash) || resp.try(:to_h) }
```

Comment:
If this looks paranoid, it's because I wanted it to be as general-purpose as possible. Given any error (Faraday or not), it will try to extract a response hash. That leaves room for us to author our own errors that conform somewhat to Faradays (which is what we might be doing already?)
Also, I noticed we were already using `to_hash`, but I noticed in _Gemfile.lock_ we're using Faraday *0.17.0*.
`Response` s in that version of Faraday don't have a `to_hash` method (it wasn't introduced until `1.0.0`).
Therefore, to be maximally, safe, I try to use the `status`, `body`, and `headers` first (which are in all versions).
https://www.rubydoc.info/gems/faraday/0.17.0/Faraday/Response
https://www.rubydoc.info/gems/faraday/Faraday/Response

Reply:
Should we be doing a Faraday Response object instead of just a hash if we are being paranoid?

Reply:
I'm not sure if it would be serialized when handed to sentry, if that makes sense.
maybe tacking it on in addition to the hash would be good